source/MonoGame.Extended.Graphics/Batcher2D.cs

```diff
@@ -132,7 +132,9 @@ private void BuildSortedIndices()
         /// <param name="drawCall">The draw call information.</param>
         protected override void InvokeDrawCall(ref DrawCallInfo drawCall)
         {
-            GraphicsDevice.DrawIndexedPrimitives(drawCall.PrimitiveType, 0, drawCall.StartIndex, drawCall.PrimitiveCount);
+            int minVertexIndex = 0;
+            int numVertices = _vertexBuffer.VertexCount;
+            GraphicsDevice.DrawIndexedPrimitives(drawCall.PrimitiveType, 0, minVertexIndex, numVertices, drawCall.StartIndex, drawCall.PrimitiveCount);
```

Comment:
The short version of GraphicsDevice.DrawIndexedPrimitives() is an MG extension.
It doesn't exist in FNA.

The short version was added because SharpDX (and presumably DX11?) didn't make use of minVertexIndex and numVertices and neither did glDrawElements(). However, FNA uses glDrawRangeElements() in OpenGL.

An alternative is to add an extension method like Nez does, because I think the original XNA method emits an Obsolete warning in MG. The fix here maps the entire VertexBuffer range. To get any performance benefit of glDrawRangeElements(), we would have to pass the information of vertices range in DrawCallInfo.

Reply:
Adding an extension method would be fine, as mentioned in other review comments.  Would go in in the **MonoGame.Extended** project. Create a new file in `/Graphics/GraphicsDevice.Extensions.cs`

Since is is dealing with a custom batcher though, priority should be on performance.  Could you speak a little more on that or if it's something that can be added that works with MG, FNA, and Kni?

Reply:
That's a taught one.

official documentation says:
 - If the range include vertices that are not referenced by the index buffer,  the implementation may partially process unused vertices, reducing performance from what could be achieved with an optimal index set.
 - It is an error for indices to lie outside the range [start,end]. the result is unspecified.

stackoverflow says:
 - glDrawRangeElements is not faster when using buffers. That was for client/user memory.
 - The drivers ignore the range.
 - Newer Draw methods in 4.5 don't have Range analogs.
 
 I 've tested glDrawRangeElements  in one of my benchmarks and didn't see any noticeable change.
 
Our options are:
 - call the full method everywhere, with #pragma to ignore the obsolete method from MonoGame.
 - use #if/#else to call the full method for FNA and pass the entire _vertexBuffer count for range, or pass the exact range down through DrawCallInfo struct.
 - Add an extension method for FNA. In that case we don't have access to _vertexBuffer , and to follow the official documentation we'd have to call GraphicsDevice.GetVertexBuffers() to get the vertices count.
 
Likely passing an optimal range will have zero benefit.
Even passing random values for range will have no side effects. Nez is passing a range [0, primitiveCount*2] which technically is incorrect. Maybe it works with the rest of Nez code, or it simply doesn't matter.

https://registry.khronos.org/OpenGL-Refpages/gl4/html/glDrawRangeElements.xhtml
https://github.com/prime31/Nez/blob/0c272de7772926742c59c196d55f0987599014ff/Nez.Portable/Utils/MonoGameCompat.cs#L23

Reply:
Changed to a GraphicsDevice.Extension method.